### ISSUE TITLE
[symfony/property-access] v.3.4.19 add/remove not used after changes in PropertyAccessor

### ISSUE THREAD

**Issue by jifer** [symfony/symfony#29485]
This change made in version 3.4.19 by @nicolas-grekas created bug breaking few forms in our project.

https://github.com/symfony/property-access/commit/15c463a508a2fefde439fd4291fd9140f4cdc5c5#diff-5ef851f935a573b3b27e674ac708651b

If model/entity contains 2 methods `addX` and `removeX` everything is ok. But if model/entity contains all 3 methods (`addX`, `removeX` and `setX`) 'set' method will be used instead of 'add' as it is higher in `if/elseif` hierarchy.
Before change (3.4.18) script was determining if model/entity has add/remove methods before entering `if/elseif` section.

**Deleted code:**
```
        if (\is_array($value) || $value instanceof \Traversable) {
            $methods = $this->findAdderAndRemover($reflClass, $singulars);
             if (null !== $methods) {
                $access[self::ACCESS_TYPE] = self::ACCESS_TYPE_ADDER_AND_REMOVER;
                $access[self::ACCESS_ADDER] = $methods[0];
                $access[self::ACCESS_REMOVER] = $methods[1];
            }
        }
```
Because of lack of this code first accessor found is `set` instead of `add`. This is breaking backward compatibility and forms request handling behavior since forever.

Unit tests don't provide check for such situation. Probably this is the reason why such error crept in.

With regards.

**Comment by dmaicher** [comment]
This looks like a duplicate of https://github.com/symfony/symfony/issues/29340

So it was fixed already in 3.4-dev but was not released yet.

**Comment by xabbuh** [comment]
Indeed, but the good news is that 3.4.20 is going to fix it soon.
